src/libs/actions/Report.js

```diff
@@ -689,29 +682,27 @@ function readNewestAction(reportID, createdDate) {
 }
 
 /**
- * Sets the last read comment on a report
+ * Sets the last read timestamp on a report
  *
  * @param {String} reportID
- * @param {String} createdDate
- * @param {Number} sequenceNumber
+ * @param {String} created
  */
-function markCommentAsUnread(reportID, createdDate, sequenceNumber) {
-    const newLastReadSequenceNumber = sequenceNumber - 1;
+function markCommentAsUnread(reportID, created) {
+    // We subtract 1 millisecond so that the lastReadTimestamp is updated to just before a given reportAction's created date
+    // For example, if we want to mark a report action with timestamp 1000 unread, we set the lastReadTimestamp to 999
+    // Since the report action with timestamp 1000 will be the first with a timestamp above 999, it's the first one that will be shown as unread
+    const lastReadTimestamp = moment(created).utc().valueOf();
     API.write('MarkAsUnread',
         {
             reportID,
-
-            // We subtract 1 millisecond so that the lastRead is updated to just before this reportAction's created date
-            createdDate: DateUtils.getDBTime(moment.utc(createdDate).valueOf() - 1),
-            sequenceNumber,
+            createdDate: DateUtils.getDBTime(lastReadTimestamp),
         },
         {
             optimisticData: [{
                 onyxMethod: CONST.ONYX.METHOD.MERGE,
                 key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
                 value: {
-                    lastReadSequenceNumber: newLastReadSequenceNumber,
-                    lastReadTimestamp: Date.now(),
+                    lastReadTimestamp,
```

Comment:
Not sure if if I am missing something here but the comment talks about subtracting 1 millisecond, but we dont do it here.

Reply:
Whoops, that seems to have gotten lost in the last commit when I switched to using `moment`. Thanks for catching!